### PR TITLE
Using patch/apply on update pods.

### LINKED ISSUE #2056: Ungating pods should use patch/apply instead of update
**What happened**:

When there are new fields in the Pod API, and Kueue is using an old k8s library versions, it is possible for it to drop fields that it doesn't know about when using update.

The simple solution is to use patch or server-side-apply instead.

**What you expected to happen**:

The pod integration to be resilient to k8s upgrades.

**How to reproduce it (as minimally and precisely as possible)**:

- Use k8s 1.30, kueue v0.6.x
- Enable pod integration
- Create a pod using annotation `container.apparmor.security.beta.kubernetes.io/main`

**Anything else we need to know?**:

Found in #2029

**Environment**:
- Kubernetes version (use `kubectl version`): 1.30
- Kueue version (use `git describe --tags --dirty --always`): 0.6
- Cloud provider or hardware configuration: any
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind bug
/kind feature

#### What this PR does / why we need it:
Using patch/apply on update pods.

#### Which issue(s) this PR fixes:

Fixes #2056 

#### Does this PR introduce a user-facing change?

```release-note
Use PATCH updates for pods. This fixes support for Pods when using the latest features in Kubernetes v1.29
```

### CODE REVIEW

File: pkg/util/client/client.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+package client
+
+import (
+	"context"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func CreatePatch(before, after client.Object) (client.Patch, error) {
```
Comment (alculquicondor) [inline pkg/util/client/client.go:25]: can you remove the similar function in the `workload` package?

Comment (mbobrovskyi) [reply pkg/util/client/client.go:25]: You mean this one https://github.com/kubernetes-sigs/kueue/commit/dddbf3db029beeee63a7417b294b2393fa5f7e5c? I didn't found it.

Comment (alculquicondor) [reply pkg/util/client/client.go:25]: oh sorry, @gabesaba just experimented with it but we didn't merge it :)

**Comment by alculquicondor** [comment]
/release-note-edit
/kind bug
```release-note
Use PATCH updates for pods. This fixes support for Pods when using the latest features in Kubernetes v1.29
```

**Comment by tenzen-y** [comment]
@alculquicondor Could we cherry pick this to 0.6?

**Comment by tenzen-y** [comment]
@mbobrovskyi Could you submit a cherry pick PR manually? This cherrypicker would be helpful: https://github.com/kubernetes-sigs/kueue/blob/main/hack/cherry_pick_pull.sh

**Comment by alculquicondor** [comment]
/release-note-edit

```release-note
Pod Integration: Use PATCH updates for pods. This fixes support for Pods when using the latest features in Kubernetes v1.29
```

**Pull request merged**